linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/JSGen.scala

```diff
@@ -133,6 +133,10 @@ private[emitter] final class JSGen(val config: Emitter.Config) {
   def globalRef(name: String)(implicit pos: Position): WithGlobals[VarRef] =
     WithGlobals(VarRef(Ident(name)), Set(name))
 
+  def untrackedGlobalRef(name: String)(implicit pos: Position): WithGlobals[VarRef] =
```

Comment:
Assert that the global ref is not dangerous?

Also, de-duplicate with the code in `CoreJSLib`? (I realize this means more allocations if `WithGlobals` cannot be stack allocated, but I don't think that matters in the `CoreJSLib`).

Reply:
> Assert that the global ref is not dangerous?

Good point.

> Also, de-duplicate with the code in `CoreJSLib`? (I realize this means more allocations if `WithGlobals` cannot be stack allocated, but I don't think that matters in the `CoreJSLib`).

I tried that, but in `CoreJSLib` the assertion is stronger than what this achieves: it asserts that *all* global refs emitted by the `CoreJSLib` (including through any method that returns `WithGlobal`s, not only `globalRef`) are non-dangerous. Using this new method would be weaker.